components/ShareBike.tsx

```diff
@@ -0,0 +1,154 @@
+    const [error, setError] = useState<string | undefined>(undefined)
+    const [shareinfo, setShareinfo] = useState({
+        email: "",
+        bikeId: bike.id,
+        role: "user",
+        duration: 86400 // 86400 = 1 day but in seconds.
+    })
+
+    const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        try {
+            setError(undefined)
+            setSuccessModal(false)
+            const result = await api.createBikeSharingInvitation(shareinfo)
+            if(result.result) {
+                // TODO: success popup
+                setSuccessModal(true)
+                // console.log(result)
+            } else if(result.message){
+                setError(`${result.message}`)
+            } else {
+                setError(`${result}`)
+            }
+            
+        } catch (e) {
+            setError(`${e}`)
+        }
+    }
+
+    return (
+        <div className='shareBike'>
+            <h3>Share bike</h3>
+
+            <form onSubmit={onSubmit}>
+                <div className="inputGroup">
+                    <label htmlFor="email">Email</label>
+                    <input 
+                        className = "inputMail"
+                        type = "email" 
+                        id = "email" 
+                        name = "email" 
+                        placeholder = "Enter email" 
+                        value = { shareinfo.email } 
+                        onChange = {
+                            e => setShareinfo(v => ({
+                                ...v, email: e.target.value
+                            }))
+                        }
+                        required 
+                    />
+
+                    <p>How many days would you like to share your bike?</p>
+                    <input 
+                        className = "inputNumber"
+                        type = "range" 
+                        id = "days" 
+                        name = "days" 
+                        step = "1"
+                        min  = "1" 
+                        max = "100" // currently a set max for the slider for higher numbers use the input box.
+                        onChange = { 
+                            e => setShareinfo(v => ({
+                                ...v, duration: Number(e.target.value) * 86400
+                            }))
+                        }
+                        value = { shareinfo.duration / 86400 }
+                    />
+
+                    <input
+                        className = "inputCustom"
+                        type = "number"
+                        min = "1"
+                        onChange = { 
+                            e => setShareinfo(v => ({
+                                ...v, duration: Number(e.target.value) * 86400
+                            }))
+                        }
+                        value = { shareinfo.duration / 86400 }
+                    />
```

Comment:
Can we change this to using `<ShareDurationSlider .. />`

Reply:
If it works definitely :)